relay-profiling/src/sample.rs

```diff
@@ -95,6 +126,110 @@ impl Profile {
             frame.strip_pointer_authentication_code(addr);
         }
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
+        let mut active_ranges: HashMap<u64, Range<usize>> = HashMap::new();
+
+        for (i, sample) in self.samples.iter().enumerate() {
+            let is_active = match self.stacks.get(sample.stack_id) {
+                Some(stack) => !stack.is_empty(),
+                None => true,
+            };
+
+            if !is_active {
+                continue;
+            }
+
+            if let Some(range) = active_ranges.get_mut(&sample.thread_id) {
+                range.end = i + 1;
+            } else {
+                active_ranges.insert(sample.thread_id, i..i + 1);
+            }
+        }
+
+        self.samples = self
+            .samples
+            .drain(..)
+            .enumerate()
+            .filter(|(i, sample)| {
+                if let Some(range) = active_ranges.get(&sample.thread_id) {
```

Comment:
```rs
active_ranges.get(&sample.thread_id).map_or(false, |range| range.contains(i))
```

Actually thinking about it, this entire `filter` could be a `filter_map`, something like this:

```rs
.filter_map(|(i, sample)|
    active_ranges.get(&sample.thread_id)
        .map_or(false, |range| range.contains(i))
        .then_some(sample)
)
```

Reply:
What about 
```rust
|(i, sample)| active_ranges
                .get(&sample.thread_id)
                .is_some_and(|range| range.contains(&i)))```

Reply:
Oh that's stable now, nice!